File: example/typescript/class/singleon-constructor.ts
class Singleon {
  private constructor() {}

  private static _instance

  static getInstance(): Singleon {
    if (this._instance == null) {
      this._instance = new Singleon()
    }
    return this._instance
  }
}

// const singleonInstance = new Singleon()  // 类“Singleon”的构造函数是私有的，仅可在类声明中访问。
const singleonInstance = Singleon.getInstance()
const singleonInstance2 = Singleon.getInstance()
console.log(singleonInstance == singleonInstance2)  // true

File: example/dart/class/singleon-constructor.dart
class Singleon {
  Singleon._();
 
  static Singleon _instance;
 
  static Singleon getInstance() {
    if (_instance == null) {
      _instance = Singleon._();
    }
    return _instance;
  }
}

main(List<String> args) {
  // var singleonInstance = new Singleon(); // The class 'Singleon' doesn't have a default constructor.
  var singleonInstance = Singleon.getInstance();
  var singleonInstance2 = Singleon.getInstance();
  print(singleonInstance == singleonInstance2); // true
}